src/core/adapters/linear.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * Linear adapter
+ *
+ * The adapter extracts ticket information from the markup. Note that linear
+ * obfuscates classes, so the DOM selectors are probably not reliable (for one,
+ * they are locale dependent, but at this time Linear is only available in
+ * English).
+ *
+ */
+
+import { match } from "micro-match";
+
+import type { TicketData } from "../types";
+import { $text } from "./dom-helpers";
+
+async function scan(url: URL, document: Document): Promise<TicketData[]> {
+  if (url.hostname !== "linear.app") return [];
+
+  const { id, issueType } = match("/:team/:issueType/:id/:slug", url.pathname);
+  const title = $text('[title="Edit issue title"]', document);
```

Comment:
This is indeed a bit fragile 🧱

Reply:
The markup looks like this:

<img width="544" alt="Screenshot 2022-03-07 at 17 01 01" src="https://user-images.githubusercontent.com/4074/157059916-54bc0386-3faa-4ff0-a72d-61986fa780bf.png">

So I don't see a way around this other that the one I already mentioned: use the official SDK which is a client for a Graphql API (but as I said I played around with it seems a bit slow, I don't want to wait seconds to grab a branch command 🙃). Ofc using an API client would be much more reliable and maintainable, so if that's your preference I could explore that alternative.